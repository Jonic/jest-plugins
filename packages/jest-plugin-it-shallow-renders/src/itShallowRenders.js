// Libraries
import {shallow} from 'enzyme';
import toJSON from 'enzyme-to-json';


const itShallowRenders = (render) => {
  it('shallow renders without error', () => {
    expect(render).not.toThrow();
  });

  it('shallow renders snapshot', () => {
    const rendered = render();
    const element = toJSON(shallow(rendered));

    expect(element).toMatchSnapshot();
  });
};


export default itShallowRenders;
