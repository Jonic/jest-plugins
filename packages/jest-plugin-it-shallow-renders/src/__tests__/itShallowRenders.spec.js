// Libraries
import React from 'react';

// Modules
import itShallowRenders from '../itShallowRenders';


// Test Components
const View = () => <div><div /></div>;
const Text = () => <div><div /></div>;


/* eslint-disable no-undef */
describe('itShallowRenders', () => {
  describe('View', () => {
    itShallowRenders(() => <View />);
  });

  describe('Text', () => {
    itShallowRenders(() => <Text />);
  });
});
