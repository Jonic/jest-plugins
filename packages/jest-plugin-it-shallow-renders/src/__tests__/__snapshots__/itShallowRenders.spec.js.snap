// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`itShallowRenders Text shallow renders snapshot 1`] = `
<View>
  <div />
</View>
`;

exports[`itShallowRenders View shallow renders snapshot 1`] = `
<div>
  <div />
</div>
`;
